sw/device/lib/dif/dif_kmac.c

```diff
@@ -595,17 +595,17 @@ dif_result_t dif_kmac_absorb(const dif_kmac_t *kmac,
   // will not be byte swapped in big-endian mode.
   const uint8_t *data = (const uint8_t *)msg;
   for (; len != 0 && ((uintptr_t)data) % sizeof(uint32_t); --len) {
-    mmio_region_write8(kmac->base_addr, KMAC_MSG_FIFO_REG_OFFSET,
-                       *data++);
+    mmio_region_write8(kmac->base_addr, KMAC_MSG_FIFO_REG_OFFSET, *data++);
   }
   for (; len >= sizeof(uint32_t); len -= sizeof(uint32_t)) {
+    uint32_t current_word = 0;
+    memcpy(&current_word, data, sizeof(uint32_t));
```

Comment:
I don't remember intricate details about strict aliasing rules, but this seems correct to me.

Reply:
I think maybe `write_32` in `sw/device/lib/base/memory.h` is more appropriate, since we know that the buffer is aligned at this point.

Reply:
The`write_32` does seem to fit in this case:
```
* @param value the value to store.
 * @param ptr a word-aligned pointer pointed to at least four bytes of memory.
 */
inline void write_32(uint32_t value, void *ptr) {
  ptr = __builtin_assume_aligned(ptr, alignof(uint32_t));
  __builtin_memcpy(ptr, &value, sizeof(uint32_t));
}
```

Reply:
Yeah, you are right, I think you need `inline uint32_t read_32(const void *ptr)` instead.

Reply:
` read_32` is the appropriate function.